rcl/test/mocking_utils/patch.hpp

```diff
@@ -0,0 +1,352 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```

Comment:
Comment for @hidmic as well. Since this file is going to be copied to several packages, I think it would be good to have a main copy checked in somewhere, perhaps the mimick_vendor package. Then in this file that's copied, it should contain a link to the original so any changes future contributors want to make can be made back to the original as well.

Reply:
I discussed that with @hidmic previously, is similar as the repeated utils used for bad allocators used in `rcutils` and `rcl` that are repeated code. We have to decide where to put all that repeated code (testing_utils) ?

However, for the moment, I think it shouldn't block this PR.

Reply:
I agree that a central location is a larger consideration, but this copy is not peer-reviewed, whereas the one over in rcutils was. So I guess my suggestion is just to add link to that file in the comments here, so curious people can follow the discussion around this file.

Reply:
I agree to the suggestion of adding the link to the peer reviewed copy, see 0e986e3